walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,97 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.List;
+import java.util.LinkedList;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
+  /**
+   * This method find the available times to schedule a meeting. It finds times that
+   * all required attendees can make, and also tries to include optional attendees.
+   * However, it either includes all the optional attendees or none of them.
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    int startOfDay = TimeRange.START_OF_DAY;
+    int endOfDay = TimeRange.END_OF_DAY;
+    int duration = (int) request.getDuration();
+    
+    if (startOfDay + duration > endOfDay) {
+        return Arrays.asList();
+    }
+
+    if (events.isEmpty() || (request.getAttendees().isEmpty() && request.getOptionalAttendees().isEmpty())) {
+        return Arrays.asList(TimeRange.fromStartEnd(startOfDay, endOfDay, true));
```

Comment:
don't return mutable collections, `return ImmutableList.of(...)` same comment everywhere.